doc/types.md

```diff
@@ -0,0 +1,59 @@
+# Types
+
+Every programming language needs a foundation in sound type theory.
+Lean has a rich provable type theory that is built on type universes.

```

Comment:
I'm not sure what it means for a type theory to be provable.

Reply:
Ok, I'm just trying to say something interesting, how would you describe the Lean type system and how it is different from other programming languages?

Reply:
I understand, and to be honest I don't think there's one good answer! But other parts of the manual already prove motivation for the type system, for example [here](https://leanprover.github.io/lean4/doc/decltypes.html). So I think it may well be okay to say nothing here and just point the reader at those other sections.

Reply:
Btw, since universes are mostly a distraction for a beginner, if the language manual is meant to be somewhat of an introduction rather than an exhaustive reference, and if `types.md` is meant to be the entrypoint into documentation about types, I would put the universe syntax in one of the more detailed type theory sections.